components/eventsrv/src/client.rs

```diff
@@ -32,16 +33,16 @@ use std::io::prelude::*;
 use std::path::Path;
 use std::thread::sleep;
 use std::time::Duration;
-
-use zmq::{Context, PUSH};
+use rand::{thread_rng, Rng};
+use zmq::{Context, PUB};
 use protobuf::Message;
 
 use message::event::{EventEnvelope, EventEnvelope_Type};
 
 fn main() {
     let ctx = Context::new();
-    let socket = ctx.socket(PUSH).unwrap();
```

Comment:
This is better for sure because the client needs to connect to the server, not the other way around. However, I think you maybe want a `DEALER/REP` socket here. A `PUB/SUB` socket will send our message to *every* `SUB` socket that we're connected to and what we want to do is fair-queue among them